### ISSUE TITLE
history={browserHistory} is still creating hashes

### ISSUE THREAD

**Issue by MattMcFarland** [rackt/react-router#2812]
Been stumped for about an hour, cant seem to change the history mode, it is always using hash no matter what.  I'm not getting error messages either.  

the code:
```javascript
import React from 'react';
import {render} from 'react-dom';
import { IndexRoute, Router, Route, browserHistory } from 'react-router';

import { Proto, NoMatch, List, Single, Layout } from './components';

render((
  <Router history={browserHistory}>
    <Route path="/" component={Layout}>
      <IndexRoute component={Proto} />
      <Route path="list" component={List}/>
      <Route path="/code/:id" component={Single}/>
      <Route path="*" component={NoMatch}/>
    </Route>
  </Router>
), document.getElementById('main'));
```

and I keep seeing the stupid `/#/` - over and over it is so frustrating.  Not sure what is going on??  is there anything I'm missing??

THanks!
Matt

**Comment by toomastahves** [comment]
try importing it from history library

import browserHistory  from 'history/lib/createBrowserHistory';
let history = browserHistory();

**Comment by m6g6a** [comment]
You can install the npm History package and make something like this:
```javascript
import { createHistory, useBasename } from 'history'

const history = useBasename(createHistory)({
	basename: '/'
})

render(
	<Router history={history}>
        ....
)
```

**Comment by timdorr** [comment]
@m6g6a That will result in deprecation warnings. You should use the `useRouter` wrapper from react-router.

@MattMcFarland I'm not seeing this happening. Anything else going on with your setup? Have you checked for history usage elsewhere in your project? You don't need the history dependency now, so make sure it's remove (or is updated to 1.17.0).

**Comment by taion** [comment]
You probably don't have the 2.x RC installed. Make sure you're using docs that match your installed version.

**Comment by MattMcFarland** [comment]
@talon you are right, I had the RC installed.  the funny thing is the docs dont match the docs on github.  anyway I managed to get it working using the createHistory method.  THanks!

**Comment by timdorr** [comment]
If you mean the 1.0.0 rc's, then that means you're upgrading to 1.0.3. In which case, the docs on master are for 2.0.0 (which is current in rc). Check out the 1.0.x docs: https://github.com/rackt/react-router/tree/1.0.x/docs

**Comment by MattMcFarland** [comment]
I installed @latest, but it does not have the latest docs, it stil is mentioning createBrowser

**Comment by ryanflorence** [comment]
latest is 1.0.3

**Comment by m6g6a** [comment]
actually @latest is a user account 😈

**Comment by MattMcFarland** [comment]
Ok when I do latest I see I am running

```javascript
 "react-router": "^2.0.0-rc4",
```

which is messed up because it tells me to use `browserHistory` in an error message, but `browserHistory` is not included and is undefined.

The `docs` say to use `browserHistory` as well in some places, and in other places say to use the older method.

Maybe npm v3 treats latest tag different?

**Comment by daveschumaker** [comment]
Pardon me while I conduct some thread necromancy here...

@MattMcFarland -- I was recently having the same issue with my project and solved it by deleting / reinstalling my `node_modules` folder.

I was using `hashHistory` from from the history library. I decided to change things up, so I removed it and imported `browserHistory` from react-router:

```
render((
  <Router history={browserHistory}>
    <Route path="/" component={Layout}>
        ...
    </Route>
  </Router>
), document.getElementById('main'));
```

Just like you, I still kept seeing `/#/` in my paths. It made no sense and was driving me crazy. I went through my project and made sure there were no references to `history` or `hashHistory`. But this still kept happening.

Finally, I just deleted the `node_modules` folder in my project and re-ran `npm install`. Things now work fine. Hah!
